peekingduck/pipeline/utils/bbox/transforms.py

```diff
@@ -172,21 +393,57 @@ def xyxy2xyxyn(inputs: np.ndarray, height: float, width: float) -> np.ndarray:
         array([[0.005, 0.02, 0.15, 0.4]])
 
     Args:
-        inputs (np.ndarray): Input bounding boxes (2-d array) each with the
+        inputs (BboxType): Input bounding boxes of shape (..., 4) each with the
             format `(top left x, top left y, bottom right x, bottom right y)`.
 
     Returns:
-        (np.ndarray): Bounding boxes with the format `normalized (top left x,
-        top left y, bottom right x, bottom right y)`.
+        outputs (BboxType): Bounding boxes with the format `normalized (top left x,
+            top left y, bottom right x, bottom right y)`.
+    """
+    outputs = clone(inputs)
+    outputs = cast_int_to_float(outputs)
+
+    outputs[..., [0, 2]] = inputs[..., [0, 2]] / width
+    outputs[..., [1, 3]] = inputs[..., [1, 3]] / height
+
+    return outputs
+
+
+def xyxyn2xyxy(inputs: BboxType, height: float, width: float) -> BboxType:
+    """Converts from normalized [x1, y1, x2, y2] to [x1, y1, x2, y2].
+
+    (x1, y1) is the top left corner and (x2, y2) is the bottom right corner.
+
+    [x1, y1, x2, y2] is calculated as:
+    x1 = Normalized x1 * width
+    y1 = Normalized y1 * height
+    x2 = Normalized x2 * width
+    y2 = Normalized y2 * height
+
+    Example:
+        >>> a = xyxyn2xyxy(inputs=np.array([[0.005, 0.02, 0.15, 0.4]]), height=100, width=200)
+        >>> a
+        array([[1.0, 2.0, 30.0, 40.0]])
+
+    Args:
+        inputs (BboxType): Input bounding boxes of shape (..., 4) each with the
+            format `normalized (top left x, top left y, bottom right x, bottom
+            right y)`.
+
+    Returns:
+        outputs (BboxType): Bounding boxes with the format `(top left x, top left y,
+            bottom right x, bottom right y)`.
     """
-    outputs = np.empty_like(inputs)
-    outputs[:, [0, 2]] = inputs[:, [0, 2]] / width
-    outputs[:, [1, 3]] = inputs[:, [1, 3]] / height
+    outputs = clone(inputs)
+    outputs = cast_int_to_float(outputs)
+
+    outputs[..., [0, 2]] = outputs[..., [0, 2]] * width
+    outputs[..., [1, 3]] = outputs[..., [1, 3]] * height
```

Comment:
Oh right i missed that.

In that case, I think this can be refactored to multiplication assignment operation instead:
```
    outputs[..., [0, 2]] *= width
    ...
```

Reply:
Changed.